### PR TITLE
added validation check to see if the node is an Array

### PR DESCRIPTION
**Pull request by SrRickGrimes**
I was getting the same issue posted here.

fixes https://github.com/microsoft/kiota-typescript/issues/1151

I added a simple validation check to confirm the node is an Array.

### CODE REVIEW

**Comment by SrRickGrimes** [comment]
@microsoft-github-policy-service agree

File: packages/serialization/json/src/jsonParseNode.ts [hunk 1]
```diff
@@ -82,15 +86,17 @@ export class JsonParseNode implements ParseNode {
     return undefined;
   }
   public getCollectionOfObjectValues = <T extends Parsable>(
-    method: ParsableFactory<T>,
+    method: ParsableFactory<T>
   ): T[] | undefined => {
-    return this._jsonNode ? (this._jsonNode as unknown[])
-      .map((x) => new JsonParseNode(x))
-      .map((x) => x.getObjectValue<T>(method)) : undefined;
+    return this._jsonNode
```
Comment (baywet) [inline packages/serialization/json/src/jsonParseNode.ts:91]: this spot also needs the check

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for the contribution!
Added a comment.

File: packages/serialization/json/src/jsonParseNode.ts (hunk 1 above)
Comment (SrRickGrimes) [reply packages/serialization/json/src/jsonParseNode.ts:94]: Thank you. I added a second validation. Please let me know if something else is missing.

**Review by baywet — APPROVED** [review]
thanks for making the changes!

**Pull request merged**